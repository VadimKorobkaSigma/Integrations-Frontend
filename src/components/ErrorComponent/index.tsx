import React from 'react';
import styles from './styles.module.scss';

interface Props {
    error: string | null;
}

const ErrorComponent: React.FC<Props> = ({ error }) => {
    return error ? <h3 className={styles.message}>Error: {error}</h3> : null;
};

export default React.memo(ErrorComponent);
