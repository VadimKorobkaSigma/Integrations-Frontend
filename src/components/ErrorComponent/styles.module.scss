.message {
  color: #f1605d;
  margin-bottom: 20px;
}
