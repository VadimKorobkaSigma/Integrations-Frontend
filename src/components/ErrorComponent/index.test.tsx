import * as React from 'react';
import ErrorComponent from './index';
import { shallow } from 'enzyme';

test('ErrorComponent Component without error', () => {
    const wrapper = shallow(<ErrorComponent error={null} />);
    expect(wrapper.html()).toMatchSnapshot();
});

test('ErrorComponent Component with error', () => {
    const wrapper = shallow(<ErrorComponent error="test" />);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ error: null });
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ error: 'test2' });
    expect(wrapper.html()).toMatchSnapshot();
});
