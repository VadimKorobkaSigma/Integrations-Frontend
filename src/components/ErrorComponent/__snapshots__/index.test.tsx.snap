// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent Component with error 1`] = `"<h3>Error: test</h3>"`;

exports[`ErrorComponent Component with error 2`] = `null`;

exports[`ErrorComponent Component with error 3`] = `"<h3>Error: test2</h3>"`;

exports[`ErrorComponent Component without error 1`] = `null`;
